test/completion.cjs

```diff
@@ -16,443 +16,512 @@ describe('Completion', () => {
+                .options({
+                  foo: {describe: 'foo option'},
+                  bar: {describe: 'bar option'},
+                })
+                .completion().argv
+          );
 
-    it('avoids repeating already included options', () => {
-      const r = checkUsage(
-        () =>
-          yargs([
-            './completion',
-            '--get-yargs-completions',
-            './completion',
-            '--foo',
-            '--',
-          ])
-            .options({
-              foo: {describe: 'foo option'},
-              bar: {describe: 'bar option'},
-            })
-            .completion().argv
-      );
+          r.logs.should.include('--bar');
+          r.logs.should.not.include('--foo');
+        });
 
-      r.logs.should.include('--bar');
-      r.logs.should.not.include('--foo');
-    });
+        it('avoids repeating options whose aliases are already included', () => {
+          const r = checkUsage(
+            () =>
+              yargs([...firstArguments, './completion', '-f', '--'])
+                .options({
+                  foo: {describe: 'foo option', alias: 'f'},
+                  bar: {describe: 'bar option'},
+                })
+                .completion().argv
+          );
 
-    it('avoids repeating options whose aliases are already included', () => {
-      const r = checkUsage(
-        () =>
-          yargs([
-            './completion',
-            '--get-yargs-completions',
-            './completion',
-            '-f',
-            '--',
-          ])
-            .options({
-              foo: {describe: 'foo option', alias: 'f'},
-              bar: {describe: 'bar option'},
-            })
-            .completion().argv
-      );
+          r.logs.should.include('--bar');
+          r.logs.should.not.include('--foo');
+        });
 
-      r.logs.should.include('--bar');
-      r.logs.should.not.include('--foo');
-    });
+        it('completes short options with a single dash when the user did not already enter two dashes', () => {
+          const r = checkUsage(
+            () =>
+              yargs([...firstArguments, './completion', '']).options({
+                f: {describe: 'f option', alias: 'foo'},
+              }).argv
+          );
 
-    it('completes short options with a single dash when the user did not already enter two dashes', () => {
-      const r = checkUsage(
-        () =>
-          yargs([
-            './completion',
-            '--get-yargs-completions',
-            './completion',
-            '',
-          ]).options({
-            f: {describe: 'f option', alias: 'foo'},
-          }).argv
-      );
+          r.logs.should.include('-f');
+          r.logs.should.not.include('--f');
+        });
 
-      r.logs.should.include('-f');
-      r.logs.should.not.include('--f');
-    });
+        it('completes short options with two dashes when the user already entered two dashes', () => {
```

Comment:
I'm not quite sure I like this behavior, might be good to discourage users to use short options with two dashes?